FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs

```diff
@@ -76,32 +75,76 @@ public class AzureBlobContainerClient : IAzureBlobContainerClient
 
             _logger = logger;
 
-            _blobContainerClient = new BlobContainerClient(
+            var blobContainerClient = new BlobContainerClient(
                 connectionString,
                 containerName);
-            _dataLakeFileSystemClient = new DataLakeFileSystemClient(
+            InitializeBlobContainerClient(blobContainerClient);
+
+            BlobContainerClient = blobContainerClient;
+
+            DataLakeFileSystemClient = new DataLakeFileSystemClient(
                 connectionString,
                 containerName);
+        }
+
+        private BlobContainerClient BlobContainerClient
+        {
+            get
+            {
+                if (_blobContainerClient is null)
+                {
+                    lock (_blobContainerClientLock)
+                    {
+                        var externalTokenCredential = _credentialProvider.GetCredential(TokenCredentialTypes.External);
```

Comment:
maybe we need double check whether client is null before initializeing a new client? Similar case in line 120